### PR TITLE
NIFI-5583: Add cdc processor for MySQL referring to GTID.

### PR DESCRIPTION
**Pull request by yoshiata**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [x] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by yoshiata** [comment]
I will add unit tests later.

**Comment by yoshiata** [comment]
Unit tests added. Please review.

**Comment by mattyb149** [comment]
Do we need a separate processor for this, or can we just add GTID support to the existing one (perhaps with a property such as "Include GTID" or whatever, defaulting to false but if set to true, executes the GTID-related code)?

**Comment by yoshiata** [comment]
We need no separate processor.
I added a saparate processor because I was concerned that adding code to existing processors would make complicated.
I will change to add code to existing processors.

**Comment by yoshiata** [comment]
Fixed existing processor. Please re-review.

**Comment by yoshiata** [comment]
I have committed some fixes for property names, documentation and code comments.
Please review.

**Comment by pushpavanthar** [comment]
@mattyb149 kindly review this PR. This is much required feature.

**Comment by chainone** [comment]
@mattyb149  We also need this feature, please help merge this PR if it passes your review, thanks.

**Comment by chainone** [comment]
Any updates yet?

**Comment by yoshiata** [comment]
If there are no more change requests, I have no update.

**Comment by wangwentao** [comment]
@yoshiata Hi, this feature is very important!  PR will be merged?

**Comment by pvillard31** [comment]
Is it possible to rebase against the main branch (note that we don't use the master branch anymore) and squash all of your commits into a single one?

**Comment by yoshiata** [comment]
@wangwentao @pvillard31 Thank you for your comment. I have rebased against the main branch and squash into single commit.

File: nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java
Comment (pvillard31) [inline nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java:735]: Is it what we want here? https://github.com/shyiko/mysql-binlog-connector-java/blob/master/src/main/java/com/github/shyiko/mysql/binlog/BinaryLogClient.java#L331

Comment (yoshiata) [reply nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java:735]: @pvillard31 We need it to realize the following:

- The processor can get the binlog even if the binlog is purged on the connected host.
- When reconnecting to another host, the processor can get the binlog even if the binlog is already purged on that host.

Comment (pvillard31) [reply nifi-nar-bundles/nifi-cdc/nifi-cdc-mysql-bundle/nifi-cdc-mysql-processors/src/main/java/org/apache/nifi/cdc/mysql/processors/CaptureChangeMySQL.java:735]: Understood

**Review by pvillard31 — APPROVED** [review]

**Comment by pvillard31** [comment]
Code looks good to me, tested a flow to confirm that the existing behavior is not changed after this change, I merged the change to the main branch. I know this PR has been opened for a long time, thanks @yoshiata for your contribution!

**Comment by yoshiata** [comment]
Thank you for reviewing and merging!

**Pull request closed without merging**
